backend/translation/views.py

```diff
@@ -1,179 +1,212 @@
 from io import StringIO
 
-import requests
 import webvtt
 from django.shortcuts import get_object_or_404
 from drf_yasg import openapi
 from drf_yasg.utils import swagger_auto_schema
 from rest_framework import status
-from rest_framework.decorators import api_view
-from rest_framework.permissions import IsAuthenticatedOrReadOnly
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework.permissions import IsAuthenticated
 from rest_framework.response import Response
-from rest_framework.views import APIView
 from transcript.models import Transcript
 
 from .metadata import INDIC_TRANS_SUPPORTED_LANGUAGES
-from .models import Translation
+from .models import * 
 from .serializers import TranslationSerializer
 from .utils import get_batch_translations_using_indictrans_nmt_api, validate_uuid4
 
 
-class TranslationView(APIView):
-    permission_classes = (IsAuthenticatedOrReadOnly,)
+@swagger_auto_schema(
+    method="get",
+    manual_parameters=[
+        openapi.Parameter(
+            "transcript_id",
+            openapi.IN_QUERY,
+            description=("A string to pass the transcript uuid"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "target_lang",
+            openapi.IN_QUERY,
+            description=("A string to pass the target language of the translation"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "translation_type",
+            openapi.IN_QUERY,
+            description=("A string to pass the target language of the translation"),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+        openapi.Parameter(
+            "get_latest",
+            openapi.IN_QUERY,
+            description=(
+                "A string to pass whether to get the latest translation or not"
+            ),
+            type=openapi.TYPE_STRING,
+            required=True,
+        ),
+    ],
+    responses={
+        200: "Generates the translation for the given transcript_id and target_lang"
+    },
+)
+@api_view(["GET"])
+def retrieve_translation(request):
+    """
+    Endpoint to retrive a translation for a given transcript and language
+    """
+    
+    # Get the query params
+    transcript_id = request.query_params.get("transcript_id")
+    target_lang = request.query_params.get("target_lang")
+    get_latest = request.query_params.get("get_latest")
+    translation_type = request.query_params.get("translation_type")
+
+    # Ensure that the UUID is valid
+    if not validate_uuid4(transcript_id):
+        return Response(
+            {"error": "Invalid transcript_id."}, status=status.HTTP_400_BAD_REQUEST
+        )
```

Comment:
Is this really required? If so, shall we add this check in the transcript app as well to make it consistent? (I guess we can remove, since anyway we will return an error response if the id does not exist)

Reply:
Sure, I have removed this.